src/main/java/com/redhat/devtools/intellij/common/utils/DownloadHelper.java

```diff
@@ -317,6 +319,16 @@ private void save(InputStream source, Path destination, long length) throws IOEx
         destination.toFile().setExecutable(true);
     }
 
+    private boolean verify(Path path, String checksum) throws IOException {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] hash = digest.digest(Files.readAllBytes(path));
+            return MessageDigest.isEqual(hash, checksum.getBytes(StandardCharsets.UTF_8));
+        } catch (NoSuchAlgorithmException e) {
+            throw new IOException(e);
```

Comment:
Maybe add an error message `throw new IOException("Could not verify checksum for file " + path.toString(), e)`?